clippy_lints/src/casts/mod.rs

```diff
@@ -759,6 +760,33 @@ declare_clippy_lint! {
     "detects `as *mut _` and `as *const _` conversion"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for casts of small constant literals or `mem::align_of` results to raw pointers.
+    ///
+    /// ### Why is this bad?
+    /// Creating a pointer through casting from a small constant or alignment value is unsafe
+    /// and can lead to undefined behavior since it creates an invalid dangling pointer.
+    /// If this is intended, use `std::ptr::dangling` or `std::ptr::dangling_mut` instead.
```

Comment:
Creating a dangling pointer is perfectly safe and well defined. There's no need to call out potential UB of using the pointer since a person creating a dangling pointer like this will already know about it.